### PR TITLE
Use StackBuilder to allocate space on the stack.

### PR DESCRIPTION
**Pull request by ptersilie**

### CODE REVIEW

File: ykcompile/src/lib.rs [hunk 1]
```diff
@@ -147,8 +148,8 @@ impl TraceCompiler {
                 } else {
                     // All registers are occupied, so we need to spill the local to the stack. For
                     // now we assume that all spilled locals are 8 bytes big.
-                    let loc = Location::Stack(self.cur_stack_offset);
-                    self.cur_stack_offset += 8;
+                    let off = self.stack_builder.alloc(8, 8);
```
Comment (vext01) [inline ykcompile/src/lib.rs:151]: maybe fold these two lines together?

File: ykcompile/src/stack_builder.rs [hunk 2]
```diff
@@ -12,17 +14,22 @@ pub struct StackBuilder {
...
+        self.stack_top.try_into().unwrap()
     }
 
     /// Aligns `offset` to `align` bytes.
     fn align(&mut self, align: u64) {
         let mask = align - 1;
         self.stack_top = (self.stack_top + mask) & !mask
     }
+
+    /// Total allocated stack size.
```
Comment (vext01) [inline ykcompile/src/stack_builder.rs:29]: In bytes? We should say.

**Comment by vext01** [comment]
Two small comments.

**Comment by ptersilie** [comment]
Addressed comments.

**Comment by vext01** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed.

**Comment by ptersilie** [comment]
Whoops, forgot to format.

**Comment by vext01** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed!

**Pull request merged**